### PR TITLE
Add test coverage for unsupported marshal scenarios

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
* Pointers to non-blittable structs are not supported
* SizeParamIndex could be bogus

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
I had to change the signature of the pointer test to make this actually trigger a `MarshalDirectiveException` (I had something else when I was experimenting with this and never tried the `void` return that I put in the test).

I think this is actually a CoreCLR bug that `static extern void PointerToNonBlittableType(NonBlittable* pNonBlittable)` works without complaints and `static extern bool PointerToNonBlittableType(NonBlittable* pNonBlittable);` fails. I assume there's a duplicated check somewhere in the pile of code that is `/src/vm` for "this p/invoke doesn't require marshalling" that is incomplete. Not sure how much we care.

File: tests/src/Interop/PInvoke/Primitives/Pointer/PInvokePointerTest.csproj [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <DefineConstants>$(DefineConstants);STATIC</DefineConstants>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="PInvokePointerTest.cs" />
+  </ItemGroup>
+  <Import Project="../../../Interop.settings.targets" />
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/PInvoke/Primitives/Pointer/PInvokePointerTest.csproj]: Please change this to `<Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), Interop.settings.targets))\Interop.settings.targets" />`

File: tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/Invalid/InvalidParamIndex.csproj [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <DefineConstants>$(DefineConstants);STATIC</DefineConstants>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="InvalidParamIndex.cs" />
+  </ItemGroup>
+  <Import Project="../../../../Interop.settings.targets" />
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/Invalid/InvalidParamIndex.csproj]: Please change this to `<Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), Interop.settings.targets))\Interop.settings.targets" />`

**Comment by AaronRobinsonMSFT** [comment]
>  I assume there's a duplicated check somewhere in the pile of code that is /src/vm for "this p/invoke doesn't require marshalling" that is incomplete. 

Can you please file that as an issue?

File: tests/src/Interop/PInvoke/Primitives/Pointer/PInvokePointerTest.csproj (hunk 1 above)
Comment (MichalStrehovsky) [reply tests/src/Interop/PInvoke/Primitives/Pointer/PInvokePointerTest.csproj]: Yup, will do. Note that I copy-pasted the pattern from existing files under `tests/src/PInvoke`.

**Comment by MichalStrehovsky** [comment]
> Can you please file that as an issue?

Sure, filed #27408.

**Pull request merged**